modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala

```diff
@@ -159,6 +171,7 @@ class StateChannel[F[_]: Async: KryoSerializer: SecurityProvider: Random](
 
   private val blockAcceptance: Stream[F, Unit] = Stream
     .awakeEvery(1.seconds)
+    .evalTap(_ => semaphores.blockAcceptance.acquire)
```

Comment:
aa. right now I have an idea that you can create an fs2 pipe that will take an effect and run it inside the acquire/release flow 🤔 like `evalMap` but with semaphore:
```
evalMapLocked(s: Semaphore[F], fn: A => F[B]): ??? = 
  s.acquire >> fn.guarantee(s.release) // or sth like that
```
maybe with a better name :D